flink-libraries/flink-table/src/main/scala/org/apache/flink/table/calcite/FlinkPlannerImpl.scala

```diff
@@ -100,6 +101,7 @@ class FlinkPlannerImpl(
       assert(validatedSqlNode != null)
       val rexBuilder: RexBuilder = createRexBuilder
       val cluster: RelOptCluster = RelOptCluster.create(planner, rexBuilder)
+      cluster.setMetadataProvider(FlinkDefaultRelMetadataProvider.INSTANCE)
```

Comment:
I remembered that in #3058 you also set the provider to `RelMetadataQuery.THREAD_PROVIDERS` by calling:
```
RelMetadataQuery.THREAD_PROVIDERS.set(JaninoRelMetadataProvider.of(cluster.getMetadataProvider))
```
Why it's not needed here?